examples/run_T1CPMG.py

```diff
@@ -32,190 +32,189 @@
 case and 4 times in the second case.  for nScans = 2, the
 SpinCore will trigger 4 times in the first case and 8 times
 in the second case.
-'''
-#}}}
+"""
+# }}}
 from pylab import *
 from pyspecdata import *
 from numpy import *
-import SpinCore_pp 
+import SpinCore_pp
 import time
 from datetime import datetime
+
 fl = figlist_var()
-date = datetime.now().strftime('%y%m%d')
-output_name = 'TEMPOL_cap_probe_T1CPMG_32dBm'
-adcOffset = 42
-carrierFreq_MHz = 14.896494
-tx_phases = r_[0.0,90.0,180.0,270.0]
-amplitude = 1.0
-p90_us = 4.625765
-deadtime_us = 10.0
-repetition_us = 15e6
-deblank_us = 1.0
+# {{{importing acquisition parameters
+config_dict = SpinCore_pp.configuration("active.ini")
+nPoints = int(config_dict["acq_time_ms"] * config_dict["SW_kHz"] + 0.5)
+# }}}
+# {{{create filename and save to config file
+date = datetime.now().strftime("%y%m%d")
+config_dict["type"] = "echo"
+config_dict["date"] = date
+config_dict["cpmg_counter"] += 1
+filename = f"{config_dict['date']}_{config_dict['chemical']}_{config_dict['type']}"
+# }}}
+tx_phases = r_[0.0, 90.0, 180.0, 270.0]
 marker = 1.0
-SW_kHz = 4.0
-nPoints = 128
-acq_time_ms = nPoints/SW_kHz # ms
-deblank_us = 1.0
-
-tau_extra = 1500.0 # us, must be more than deadtime and more than deblank
+tau_extra = 1500.0  # us, must be more than deadtime and more than deblank
 pad_start = tau_extra - deadtime_us
-pad_end = tau_extra - deblank_us*2 # marker + deblank
-twice_tau = deblank_us + 2*p90_us + deadtime_us + pad_start + acq_time_ms*1e3 + pad_end + marker
-tau_us = twice_tau/2.0
-
-nScans = 12
-nEchoes = 64
+pad_end = tau_extra - deblank_us * 2  # marker + deblank
+twice_tau = (
+    deblank_us
+    + 2 * p90_us
+    + deadtime_us
+    + pad_start
+    + acq_time_ms * 1e3
+    + pad_end
+    + marker
+)
+tau_us = twice_tau / 2.0
 phase_cycling = True
 if phase_cycling:
     nPhaseSteps = 2
 if not phase_cycling:
-    nPhaseSteps = 1 
-data_length = 2*nPoints*nEchoes*nPhaseSteps
+    nPhaseSteps = 1
+data_length = 2 * nPoints * nEchoes * nPhaseSteps
 # NOTE: Number of segments is nEchoes * nPhaseSteps
-vd_list = r_[5e1,5.8e5,9e5,1.8e6,2.7e6,
-        3.6e6,4.5e6,5.4e6,6.4e6,7.2e6,10e6]
-for index,val in enumerate(vd_list):
+vd_list = r_[5e1, 5.8e5, 9e5, 1.8e6, 2.7e6, 3.6e6, 4.5e6, 5.4e6, 6.4e6, 7.2e6, 10e6]
```

Comment:
phase cycling components and vdlist

Reply:
wasn't sure how specific this vdlist was so I just left it with original values as I'm unsure what it should look like in a T1CPMG experiment